### PR TITLE
Don't check for updates in demo mode

### PR DESCRIPTION
**Pull request by danirabbit**

### CODE REVIEW

File: src/Backends/SystemUpdate.vala [hunk 1]
```diff
@@ -65,6 +65,10 @@ public class SettingsDaemon.Backends.SystemUpdate : Object {
     }
 
     public async void check_for_updates (bool force, bool notify) throws DBusError, IOError {
+        if (Utils.is_running_in_demo_mode ()) {
```
Comment (ryonakano) [inline src/Backends/SystemUpdate.vala:68]:
```suggestion
        if (SettingsDaemon.Utils.is_running_in_demo_mode ()) {
```

This should fix the build error

Comment (danirabbit) [reply src/Backends/SystemUpdate.vala]: Oops thanks, I think I fixed this locally and forgot to push :sweat_smile:

**Review by ryonakano — COMMENTED** [review]
Ah, we're also missing `src/Utils/SessionUtils.vala` in meson.build. That's why the build still failing.

**Comment by danirabbit** [comment]
@ryonakano sorry about the sloppy branch. Should be fixed now

**Review by ryonakano — APPROVED** [review]
Works fine and makes sense.

This would be a separated PR, but Refresh button in the System plug now does nothing because of this change. Maybe we need to add a new definition to `PkUtils.CurrentState` and use it for the sensitivity of the button.

Screenshot from 2024-10-24 20-05-28

**Comment by danirabbit** [comment]
@ryonakano Ah yeah okay, added also a check for `force` so that the refresh button should still work. Then we can consider doing something different with the UI in demo mode separately :)

**Review by ryonakano — APPROVED** [review]
Confirmed clicking the refresh button checks for update in the demo mode.

**Pull request merged**